ktor-client/ktor-client-plugins/ktor-client-logging/jvm/src/io/ktor/client/plugins/logging/LoggerJvm.kt

```diff
@@ -16,11 +17,37 @@ public actual val Logger.Companion.DEFAULT: Logger
     }
 
 /**
- * Android [Logger]: breaks up long log messages that would be truncated by Android's max log
- * length of 4068 characters
+ * Android [Logger]: Logs to the Logcat on Android if the SLF4J provider isn't found.
+ * Otherwise, uses the [Logger.Companion.DEFAULT].
+ * Breaks up long log messages that would be truncated by Android's max log
+ * length of 4068 characters.
  */
 public val Logger.Companion.ANDROID: Logger
-    get() = MessageLengthLimitingLogger()
+    get() {
+        val logger = Logger.DEFAULT
+
+        val logClass = try {
+            Class.forName("android.util.Log")
+        } catch (_: ClassNotFoundException) {
+            return MessageLengthLimitingLogger(delegate = logger)
+        }
+
+        if (LoggerFactory.getILoggerFactory() !is NOPLoggerFactory) {
+            return MessageLengthLimitingLogger(delegate = logger)
+        }
+
+        return MessageLengthLimitingLogger(delegate = LogcatLogger(logClass))
+    }
+
+private class LogcatLogger(private val logClass: Class<*>) : Logger {
+    private val tag = "Ktor Client"
+    override fun log(message: String) {
+        runCatching {
+            val method = logClass.getDeclaredMethod("i", String::class.java, String::class.java)
```

Comment:
please consider extracting this to the class level scope. I would also add fallback